cmake/CMakeLists.txt

```diff
@@ -199,13 +198,23 @@ if(onnxruntime_USE_NNAPI_BUILTIN AND onnxruntime_USE_NNAPI_DNNLIBRARY)
   message(FATAL_ERROR "Please use only one of onnxruntime_USE_NNAPI_BUILTIN, onnxruntime_USE_NNAPI_DNNLIBRARY")
 endif()
 
-if(onnxruntime_DISABLE_RTTI)
+set(onnxruntime_DISABLE_RTTI AUTO CACHE STRING "Disable RTTI")
+set_property(CACHE onnxruntime_DISABLE_RTTI PROPERTY STRINGS AUTO ON OFF)
+if(onnxruntime_DISABLE_RTTI STREQUAL "AUTO")
+  add_compile_definitions($<$<NOT:$<CONFIG:Debug>>:ORT_NO_RTTI> $<$<NOT:$<CONFIG:Debug>>:GOOGLE_PROTOBUF_NO_RTTI>)
+  if(MSVC)
+    # Disable RTTI and turn usage of dynamic_cast and typeid into errors
+    add_compile_options("$<$<AND:$<NOT:$<CONFIG:Debug>>,$<COMPILE_LANGUAGE:CXX>>:SHELL:/GR- /we4541>")
+  else()
+    add_compile_options("$<$<AND:$<NOT:$<CONFIG:Debug>>,$<COMPILE_LANGUAGE:CXX>>:-fno-rtti>")
+  endif()
+elseif(onnxruntime_DISABLE_RTTI STREQUAL "ON")
```

Comment:
The old setup of having onnxruntime_DISABLE_RTTI being set via build.py was simple and easy to understand.

What's the benefit of these changes? Seems like it's trying to automatically do things based on whether it's a debug build or not, but afaik we've never discussed wanting to automatically disable rtti for all non-debug builds. Not sure if we want that or not.